pytorch_lightning/trainer/training_loop.py

```diff
@@ -45,6 +45,8 @@ def __init__(self, trainer):
         self.automatic_optimization = True
         self._curr_step_result = None
         self._cur_grad_norm_dict = None
+        self._multiple_trainloader_mode = multiple_trainloader_mode
+        self.trainer._multiple_trainloader_mode = multiple_trainloader_mode
```

Comment:
is it required on the trainer?

Reply:
Yes. So that `_multiple_trainloader_mode` can be used in `TrainerDataLoadingMixin`. But `self._multiple_trainloader_mode = multiple_trainloader_mode` is not used, we can remove it. Maybe it's more reasonable to initialize `_multiple_trainloader_mode` directly in `Trainer` or `DataConnector`?

Reply:
I think with the current approach this flag should be read-only in trainer. So maybe we could make it a property forwarding from train loop?

Reply:
It's a good idea for me.